packages/backend-deployer/src/cdk_deployer.ts

```diff
@@ -56,7 +56,10 @@ export class CDKDeployer implements BackendDeployer {
       }
     }
 
-    // first synth with the backend definition
+    // First run type checks to ensure the project builds
+    await this.invokeTsc(deployProps);
```

Comment:
This will fail when using the generated `env` symbol in function definitions because it is only generated after synth runs